### PR TITLE
r/cognito_risk_configuration - new resource

### LINKED ISSUE #7007: AWS Cognito User Pool Doesn't Support Advanced Security Features
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently there doesn't appear to be advanced security settings in the Cognito User Pool:
https://www.terraform.io/docs/providers/aws/r/cognito_user_pool.html

It would be nice to have these features settable from Cognito:
https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-advanced-security.html

Things we would like are the following settings (not fully expansive):

-mode: audit, yes
-compromised_action: block, allow
-low_risk_setting: allow, optional mfa, mfa required
-low_risk_notification: on, off
-medium_risk_setting: allow, optional mfa, mfa required
-medium_risk_notification: on, off
-high_risk_setting: allow, optional mfa, mfa required
-high_risk_notification: on, off
-SES region
-SES email
-reply_to

### New or Affected Resource(s)

https://www.terraform.io/docs/providers/aws/r/cognito_user_pool.html

* aws_cognito_user_pool

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-advanced-security.html

* #0000

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #7007

Output from acceptance testing:

```
$ make testacc TESTS=TestAccCognitoIDPRiskConfiguration_ PKG=cognitoidp
--- PASS: TestAccCognitoIDPRiskConfiguration_disappears_userPool (29.84s)
--- PASS: TestAccCognitoIDPRiskConfiguration_compromised (38.48s)
--- PASS: TestAccCognitoIDPRiskConfiguration_client (41.34s)
--- PASS: TestAccCognitoIDPRiskConfiguration_exception (63.75s)
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccCognitoIDPRiskConfiguration_' PKG=cognitoidp ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/cognitoidp/... -v -count 1 -parallel 2  -run=TestAccCognitoIDPRiskConfiguration_ -timeout 180m
=== RUN   TestAccCognitoIDPRiskConfiguration_exception
=== PAUSE TestAccCognitoIDPRiskConfiguration_exception
=== RUN   TestAccCognitoIDPRiskConfiguration_client
=== PAUSE TestAccCognitoIDPRiskConfiguration_client
=== RUN   TestAccCognitoIDPRiskConfiguration_compromised
=== PAUSE TestAccCognitoIDPRiskConfiguration_compromised
=== RUN   TestAccCognitoIDPRiskConfiguration_disappears_userPool
=== PAUSE TestAccCognitoIDPRiskConfiguration_disappears_userPool
=== CONT  TestAccCognitoIDPRiskConfiguration_exception
=== CONT  TestAccCognitoIDPRiskConfiguration_compromised
--- PASS: TestAccCognitoIDPRiskConfiguration_compromised (19.02s)
=== CONT  TestAccCognitoIDPRiskConfiguration_client
--- PASS: TestAccCognitoIDPRiskConfiguration_exception (30.63s)
=== CONT  TestAccCognitoIDPRiskConfiguration_disappears_userPool
--- PASS: TestAccCognitoIDPRiskConfiguration_client (18.05s)
--- PASS: TestAccCognitoIDPRiskConfiguration_disappears_userPool (12.89s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp	47.173s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

File: website/docs/r/cognito_risk_configuration.html.markdown [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+* `mfa_email` - (Optional) The multi-factor authentication (MFA) email template used when MFA is challenged as part of a detected risk. See notify email type below.
+* `no_action_email` - (Optional) The email template used when a detected risk event is allowed. See notify email type below.
+* `from` - (Optional) The email address that is sending the email. The address must be either individually verified with Amazon Simple Email Service, or from a domain that has been verified with Amazon SES.
+* `reply_to` - (Optional) The destination to which the receiver of an email should reply to.
+* `source_arn` - (Required) The Amazon Resource Name (ARN) of the identity that is associated with the sending authorization policy. This identity permits Amazon Cognito to send for the email address specified in the From parameter.
+
+##### notify email type
+
+* `html_body` - (Required) The email HTML body.
+* `block_email` - (Required) The email subject.
```
[the remark this answers is not in the document]
Comment (TakumiHaruta) [reply website/docs/r/cognito_risk_configuration.html.markdown:52]: Create PR #25645

File: website/docs/r/cognito_risk_configuration.html.markdown [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+* `block_email` - (Required) The email text body.
```
[the remark this answers is not in the document]
Comment (TakumiHaruta) [reply website/docs/r/cognito_risk_configuration.html.markdown:53]: Create PR #25645

**Pull request merged**